experimental/examples/bubblebench/sharedtree/src/proxy/tree.ts

```diff
@@ -72,13 +72,13 @@ export const TreeObjectProxy = <T extends Object>(
         // },
     });
 
-export class TreeArrayProxy<T> implements IArrayish<T> {
+export class TreeArrayProxy<T> implements IArrayish<Serializable<T>> {
```

Comment:
As far as I can tell, it returns a new type with some of the members which it deems not serializable removed.

Thus we are now saying that this tree acts like an array of the serializable part of T. Before it claimed that the non-serializable portions of the data will be kept which was wrong, and failed to build in newer typescript where implicit conversions from the Serializable<SomeType> to SomeType occurred.

Specifically, methods like `pop(): Serializable<T> | undefined` no longer are considered implementations of `pop(): T | undefined`. Typescript is now rejecting it, since not all Serializable<T> extend T. This seems like an improvement in the type-checking.

It also rejects the old push method, which was actually ok (since all T are Serializable<T>), so it seems like this more strict type checking also eliminates some cases we would ideally like it to permit (we are now closer to sound, and further from complete: a classic tradeoff)

The code using the tree is unchanged since for its specific value if T, Serializable<T> and T are equal (which we expect to be true for most users)